src/tests/grid/test_grids.cc

```diff
@@ -427,55 +427,70 @@ CASE( "test_structured_from_config" ) {
-        pointsLonLat.push_back( ll );
-        g->projection().lonlat2xy(ll);
-        EXPECT_APPROX_EQ(ll, pointsXY[i], 1e-11);
-        ++i;
-    }
-    i = 0;
-    for ( auto xx : g.xy() ) {
-       g->projection().xy2lonlat(xx);
-       EXPECT_APPROX_EQ(xx, pointsLonLat[i], 1e-11);
-       ++i;
-    }
-
-    EXPECT(pointsLonLat.size() == 6* resolution* resolution +2);
-
-    if (resolution == 2) {
-        const double rpi     = 2.0 * asin( 1.0 );
-        const double rad2deg = 180.0 / rpi;
-        double cornerLat = rad2deg * atan(sin(rpi/4.0));
-        double tolerance = 1e-13;
-        // Expected latitudes/longitude per tile
-        std::vector<std::pair<double, double>> expectedLatLon{
-            {-cornerLat, 315.0}, {-45.0, 0.0}, {0.0, 315.0}, {0.0,0.0}, {cornerLat, 315.0},
-            {-cornerLat, 45.0}, {-45.0, 90.0},  {-cornerLat, 135.0}, {0.0, 45.0}, {0.0, 90.0},
-            {cornerLat, 45.0}, { 45.0, 90.0}, { 45.0, 0.0}, { 90, 0.0},
-            {cornerLat, 135.0}, {0.0, 135.0}, { 45.0, 180.0}, { 0.0, 180.0},
-            {cornerLat, 225.0},  {0.0, 225.0}, {45.0, 270.0}, { 0.0, 270.0},
-            {-cornerLat, 225.0}, {-45.0, 180.0}, {-45.0, 270.0}, {-90.0, 0.0}
-        };
-
-        for (std::size_t jn = 0; jn < g.size(); ++jn) {
-            Log::info() << " cube:: global index :" << jn+1 <<  " "
-              << "actual/expected latitude :"  << pointsLonLat[jn].lat() <<  " " << expectedLatLon[jn].first << " "
-              << "actual/expected longitude :"  << pointsLonLat[jn].lon() <<  " " << expectedLatLon[jn].second << std::endl;
-            EXPECT(std::abs(pointsLonLat[jn].lat() - expectedLatLon[jn].first) <  tolerance);
-            EXPECT(std::abs(pointsLonLat[jn].lon() - expectedLatLon[jn].second) <  tolerance);
+CASE( "test_cubedsphere" ) {
+    int resolution(2);
+    std::vector<std::string> grid_names{"CS-EA-" + std::to_string(resolution),
+                "CS-ED-" + std::to_string(resolution)};
+
+    for (std::string & s : grid_names) {
+        Grid grid{ s };
+        EXPECT(grid);
+        std::vector<PointLonLat> pointLonLats_from_XY;
+        std::vector<PointXY> pointXYs;
+        std::vector<PointLonLat> pointLonLats;
+        std::vector<PointLonLat> pointXYs_from_LonLat;
+
+        for ( auto crd : grid.xy() ) {
+            pointXYs.push_back(crd);
+            grid->projection().xy2lonlat(crd);
+            pointLonLats_from_XY.push_back(crd);
+        }
+        for ( auto crd : grid.lonlat() ) {
+            pointLonLats.push_back(crd);
+            grid->projection().lonlat2xy(crd);
+            pointXYs_from_LonLat.push_back(crd);
+        }
+        EXPECT(pointLonLats.size() == 6* resolution * resolution +2);
+        EXPECT(pointXYs.size() == 6* resolution * resolution +2);
+        EXPECT(grid.size() == 6* resolution * resolution +2);
+
+        // Note that with nodal points on the cubed-sphere
+        if (resolution == 2) {
+            const double rpi     = 2.0 * asin( 1.0 );
```

Comment:
You can use
```
constexpr double rpi = M_PI;
constexpr double rad2deg = 180. / rpi;
```
where ```M_PI``` is defined in header ```<cmath>```
Also use ```std::asin``` and ```std::atan```

Reply:
Done